pkg/odo/cli/list/services/services.go

```diff
@@ -0,0 +1,198 @@
+	%[1]s --all-namespaces
+	%[1]s -A
+
+	# List the bindable Operator backed services in JSON format
+	%[1]s -o json
+	%[1]s --all-namespaces -o json
+	%[1]s -A -o json`)
+
+	listLongDesc = ktemplates.LongDesc(`
+	List the bindable Operator backed services that could bound to the odo component 
+`)
+)
+
+type ServiceListOptions struct {
+	// clientset
+	clientset *clientset.Clientset
+	// context
+	*genericclioptions.Context
+	// working directory
+	contextDir string
+	// flags
+	namespaceFlag     string
+	allNamespacesFlag bool
+}
+
+var _ genericclioptions.Runnable = (*ServiceListOptions)(nil)
+var _ genericclioptions.JsonOutputter = (*ServiceListOptions)(nil)
+
+func (o *ServiceListOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+func (o *ServiceListOptions) Complete(cmdline cmdline.Cmdline, _ []string) error {
+	var err error
+	o.contextDir, err = os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	o.Context, err = genericclioptions.New(genericclioptions.NewCreateParameters(cmdline))
+	if err != nil {
+		return err
+	}
+
+	if o.namespaceFlag == "" && !o.allNamespacesFlag {
+		o.namespaceFlag = o.GetProject()
+	}
+
+	return err
+}
+
+func (o *ServiceListOptions) Validate() error {
+	if o.allNamespacesFlag && o.namespaceFlag != "" {
+		return errors.New("cannot use --all-namespaces and --namespace flags together")
+	}
+	return nil
+}
+
+func (o *ServiceListOptions) Run(_ context.Context) error {
+	var s *log.Status
+	if o.allNamespacesFlag {
+		s = log.Spinner("Listing bindable services from all namespaces")
+		defer s.End(false)
+	} else {
+		s = log.Spinner(fmt.Sprintf("Listing bindable services from namespace %q", o.namespaceFlag))
+		defer s.End(false)
+	}
+	services, err := o.run()
+	if err != nil {
+		return err
+	}
+	s.End(true)
+	HumanReadable(services)
+	return nil
+}
+
+func (o *ServiceListOptions) run() ([]unstructured.Unstructured, error) {
+	services := []unstructured.Unstructured{}
+
+	if o.allNamespacesFlag {
+		projects, err := o.clientset.ProjectClient.List()
+		if err != nil {
+			return services, err
+		}
+		for _, project := range projects.Items {
+			svcs, err := o.clientset.BindingClient.GetServiceInstances(project.Name)
+			if err != nil {
+				return services, err
+			}
+			for k := range svcs {
+				services = append(services, svcs[k])
+			}
+		}
+	} else {
+		svcs, err := o.clientset.BindingClient.GetServiceInstances(o.namespaceFlag)
+		if err != nil {
+			return services, err
+		}
+		for k := range svcs {
+			services = append(services, svcs[k])
+		}
+	}
+	return services, nil
+
+}
+
+func (o *ServiceListOptions) RunForJsonOutput(_ context.Context) (out interface{}, err error) {
+	services, err := o.run()
+	if err != nil {
+		return nil, err
+	}
+
+	var servicesList []api.BindableService
+	for _, svc := range services {
+		servicesList = append(servicesList, api.BindableService{Name: svc.GetName(), Namespace: svc.GetNamespace()})
```

Comment:
Also this could be done into `run`.

To go further, you could also do the embedding into the `bindableServices` field.

The "perfect" `RunForJsonOutput` function being, IMO, this:

```
func (lo *ListOptions) RunForJsonOutput(ctx context.Context) (out interface{}, err error) {
	return lo.run(ctx)
}
```

Reply:
I liked this idea. Done.